src/org/thoughtcrime/securesms/WebViewActivity.java

```diff
@@ -127,22 +122,6 @@ public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceReque
         return super.shouldInterceptRequest(view, request);
       }
     });
-    webView.setWebChromeClient(new WebChromeClient() {
-      @Override
-      @RequiresApi(21)
-      public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, WebChromeClient.FileChooserParams fileChooserParams) {
-        if (WebViewActivity.this.filePathCallback != null) {
-          WebViewActivity.this.filePathCallback.onReceiveValue(null);
-        }
```

Comment:
I discarded this because it doesn't match the behavior of browsers, it is simulating a "nothing selected" event upon opening the chooser, which doesn't happen in real browsers, not even when the user cancels the file chooser (don't select anything)

Reply:
hm, i would leave it as is. iiuc, the even goes to previous selector, not to the current one. `WebViewActivity.this.filePathCallback.onReceiveValue(null)` is called only if there was previously a file request started that gets aborted. the 3 lines make sure, a request is always somehow ended.

the code comes from the AvancedWebView, so, unless we're really sure what we're doing better, i'd leave it and focus on shifting things in this pr. if you still think, it is wrong, please open a subsequent pr.

Reply:
but why would you send such event to the previous callback when calling a new one? the old one already received the call, also the callback is set to null every time the response is received so I think this doesn't even happens never :thinking:

Reply:
> but why would you send such event to the previous callback when calling a new one?

https://developer.android.com/reference/android/webkit/WebChromeClient#onShowFileChooser says that for canceling a request, we should pass `null` - so, if forever reason, onShowFileChooser() is triggered again before we close the dialog, we should cancel the previous callback

i agree that it should not happen normally

Reply:
ok, I reverted it, it should not hurt to have in the end